src/client/async/async.js

```diff
@@ -0,0 +1,197 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Use of this source code is governed by The MIT License.
+// See the LICENSE file for details.
+
+/**
+ * @fileoverview Fast asynchronous function execution.
+ *
+ * This package provides functions to defer execution on the main thread
+ * without using setTimeout, though setTimeout is used as a fallback in browsers
+ * that do not support other methods.  Using these methods is advantageous when
+ * one wants to schedule a callback faster than the setTimeout clamped minimum
+ * allows (e.g. when doing `setTimeout(fn, 0)`)  The clamped minimum for
+ * setTimeout is often 10ms, though when WebKit browsers are in a background
+ * tab, setTimeout calls deprioritized to execute with a 1s delay.  In these
+ * cases, this package provides an alternative.
+ *
+ * @author nicksay@google.com (Alex Nicksay)
+ */
+
+goog.provide('spf.async');
+
+goog.require('spf');
+goog.require('spf.state');
+goog.require('spf.string');
+goog.require('spf.tracing');
+
+
+
+/**
+ * Initializes fast asynchronous function execution.
+ */
+spf.async.init = function() {
+  if (!spf.state.get(spf.state.Key.ASYNC_INIT)) {
+    if (spf.async.POSTMESSAGE_SUPPORTED_) {
+      spf.async.addListener_(spf.async.handleMessage_);
+      spf.state.set(spf.state.Key.ASYNC_LISTENER, spf.async.handleMessage_);
+    }
+    spf.state.set(spf.state.Key.ASYNC_INIT, true);
+  }
+};
+
+
+/**
+ * Disposes fast asynchronous function execution.
+ */
+spf.async.dispose = function() {
```

Comment:
I'm more inclined to make this like a normal setTimeout and have it always fire callbacks, which I think would mean having no init/dispose cycle but rather a persistent event handler that is replaced only if needed (not unlike for document.readyState in main.js).